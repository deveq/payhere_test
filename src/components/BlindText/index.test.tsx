import {render, screen} from '@testing-library/react';
import BlindText from './index';

describe('<BlindText />', () => {
    const text = 'Button Test';

    it('컴포넌트가 정상적으로 렌더링 되었는지 확인한다', () => {
        const {container} = render(<BlindText>{text}</BlindText>);
        const button = screen.getByText(text);
        expect(button).toBeInTheDocument();
        expect(container).toMatchSnapshot(); 
    });

    it('기본 스타일이 정상적으로 렌더링되는지 확인한다.', () => {
        render(<BlindText>{text}</BlindText>);
        const button = screen.getByText(text);
        expect(button).toHaveStyle({
            position: 'absolute',
            clip: 'rect(0 0 0 0)',
            width: '1px',
            height: '1px',
            margin: '-1px',
            overflow: 'hidden',
        })
    })
})

// const BlindText = styled.span`
//     position: absolute;
//     clip: rect(0 0 0 0);
//     width: 1px;
//     height: 1px;
//     margin: -1px;
//     overflow: hidden;
// `;