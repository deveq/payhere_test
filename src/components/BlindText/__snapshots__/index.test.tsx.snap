// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlindText /> 컴포넌트가 정상적으로 렌더링 되었는지 확인한다 1`] = `
<div>
  <span
    class="sc-bcXHqe hNOtCR"
  >
    Button Test
  </span>
</div>
`;
